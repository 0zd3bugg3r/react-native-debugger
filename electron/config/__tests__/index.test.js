import fs from 'fs';
import path from 'path';
import readConfig from '..';

const testFile = path.join(__dirname, 'config_test');

beforeAll(() => fs.existsSync(testFile) && fs.unlinkSync(testFile));

test('readConfig', () => {
  expect(readConfig(testFile)).toMatchSnapshot();

  // User custom config
  fs.writeFileSync(testFile, '{ autoUpdate: false, }');
  expect(readConfig(testFile)).toMatchSnapshot();

  // Broken config
  fs.writeFileSync(testFile, '{ autoUpdate: is_broken, }');
  expect(readConfig(testFile)).toMatchSnapshot();
});
