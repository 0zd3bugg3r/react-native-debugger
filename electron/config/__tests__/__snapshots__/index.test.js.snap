// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`openConfigFile 1`] = `
Object {
  "config": Object {
    "autoUpdate": true,
    "defaultRNPakcagerPorts": Array [
      8081,
    ],
    "defaultReactDevToolsTheme": "RNDebugger",
    "editor": "",
    "fontFamily": "monaco, Consolas, Lucida Console, monospace",
    "fontSize": 12,
    "windowBounds": Object {
      "frame": true,
      "height": 750,
      "width": 1024,
    },
  },
}
`;

exports[`readConfig 1`] = `
Object {
  "config": Object {
    "autoUpdate": true,
    "defaultRNPakcagerPorts": Array [
      8081,
    ],
    "defaultReactDevToolsTheme": "RNDebugger",
    "editor": "",
    "fontFamily": "monaco, Consolas, Lucida Console, monospace",
    "fontSize": 12,
    "windowBounds": Object {
      "frame": true,
      "height": 750,
      "width": 1024,
    },
  },
}
`;

exports[`readConfig 2`] = `
Object {
  "config": Object {
    "autoUpdate": false,
  },
}
`;

exports[`readConfig 3`] = `
Object {
  "config": Object {
    "autoUpdate": true,
    "defaultRNPakcagerPorts": Array [
      8081,
    ],
    "defaultReactDevToolsTheme": "RNDebugger",
    "editor": "",
    "fontFamily": "monaco, Consolas, Lucida Console, monospace",
    "fontSize": 12,
    "windowBounds": Object {
      "frame": true,
      "height": 750,
      "width": 1024,
    },
  },
  "isConfigBroken": true,
}
`;
